singer_sdk/plugin_base.py

```diff
@@ -168,7 +169,12 @@ def _get_package_version(package: str) -> str:
         try:
             version = metadata.version(package)
         except metadata.PackageNotFoundError:
-            version = "[could not be detected]"
+            try:
+                module = importlib.import_module(package.replace("-", "_"))
```

Comment:
@WillDaSilva great question! This has to do with the case when the package name is different to the import name used to call the `_get_package_version`. We use the Taps `cls.name` attr when calling this method, which is "The executable name of the tap or target plugin." (e.g. `tap-snowflake`). This is fine if the package was installed as `tap-snowflake`, but in the case of variants (e.g. `meltanolabs-tap-snowflake`) the package name differs from the executable and importable names (`tap-snowflake` and `tap_snowflake` respectively). So effectively `metadata.version(package)` is looking for a package that doesn't exist, throwing a `PackageNotFoundError`. However it is still possible to import `tap_snowflake` by the module name, and check its `__version__`. Its an oddity of the Singer ecosystem rather than anything specific to Python packaging 😅

Another way we could solve this is adding a `variant` or `package_name` attr to the base Tap/Target class, so the first check succeeds. However, given the Singer convention of using variant only in packaging, this works and is less work for developers 🙂